clamav/cnudie.py

```diff
@@ -23,20 +28,105 @@
 
 def scan_resources(
     resource_nodes: typing.Iterable[cnudie.iter.ResourceNode],
+    clamav_version_info: clamav.model.ClamAVVersionInfo,
+    virus_definition_age_threshold: int,
+    skip_if_virus_definition_unchanged: bool,
     oci_client: oci.client.Client,
     clamav_client: clamav.client.ClamAVClient,
-    clamav_version_info: clamav.model.ClamAVVersionInfo,
+    delivery_client: delivery.client.DeliveryServiceClient = None,
     s3_client=None,
     max_workers:int = 16,
 ) -> typing.Generator[clamav.model.ClamAVResourceScanResult, None, None]:
-    executor = concurrent.futures.ThreadPoolExecutor(max_workers=max_workers)
+
+    def _skip_scan(
+            previous_finding: delivery.model.ArtefactMetadata,
+            current_version_info: clamav.model.ClamAVVersionInfo,
+    ) -> bool:
+        # helper function to determine whether a scan may be skipped
+
+        if isinstance(previous_finding.data, dict):
+            # In this case the previous scan we have available predates the current dataclasses.
+            # It is safe to assume that this scan result is very old.
+            return False
+
+        clamav_scan_metadata = previous_finding.data.metadata
+        if (
+            skip_if_virus_definition_unchanged
+            and clamav_scan_metadata.signature_version == current_version_info.signature_version
```

Comment:
why check both version and timestamp? checking for timestamp should be enough, should it not?